File: solutions/haskell/darts/1/src/Darts.hs
module Darts (score) where

score :: Float -> Float -> Int
score x y
  | rs <= 1.0 = 10
  | rs <= 25.0 = 5
  | rs <= 100.0 = 1
  | otherwise = 0
  where 
    rs = x^2 + y^2

File: solutions/kotlin/darts/2/src/main/kotlin/Darts.kt
object Darts {

    fun score(x: Number, y: Number /* choose proper types! */): Int {
        val x = x.toDouble()
        val y = y.toDouble()
        val r2 = x*x + y*y
        return when {
            r2 > 100 -> 0
            r2 > 25 -> 1
            r2 > 1 -> 5
            else -> 10
        }
    }
}
